File: admin-web/src/services/authService.js
import apiClient from './apiClient';
import { API_CONFIG } from '../config/apiConfig';

export const authService = {
  async login(username, password) {
    const response = await apiClient.post(API_CONFIG.login, { username, password });
    const { access, refresh } = response.data;
    localStorage.setItem('access_token', access);
    localStorage.setItem('refresh_token', refresh);
    return response.data;
  },

  async getCurrentUser() {
    const response = await apiClient.get(API_CONFIG.me);
    return response.data;
  },

  logout() {
    localStorage.removeItem('access_token');
    localStorage.removeItem('refresh_token');
  },

  isLoggedIn() {
    return !!localStorage.getItem('access_token');
  },
};

export default authService;

File: delivery_app/lib/services/auth_service.dart
import 'dart:convert';
import 'dart:async';
import '../config/api_config.dart';
import 'api_service.dart';

class AuthService {
  static Future<Map<String, dynamic>> loginWithPassword(String identifier, String password) async {
    try {
      final response = await ApiService.post(
        ApiConfig.loginWithPassword,
        {'identifier': identifier, 'password': password},
      ).timeout(const Duration(seconds: 10));
      final data = jsonDecode(response.body);

      if (response.statusCode == 200) {
        if (data['user']?['role'] != 'DELIVERY_STAFF') {
          return {'success': false, 'error': 'This account is not a delivery staff account.'};
        }
        await ApiService.saveTokens(data['access'], data['refresh']);
        return {'success': true, 'user': data['user']};
      }
      return {'success': false, 'error': data['detail'] ?? 'Incorrect username or password.'};
    } on TimeoutException {
      return {'success': false, 'error': 'Connection timed out.'};
    } catch (e) {
      return {'success': false, 'error': 'Could not connect to server: $e'};
    }
  }

  static Future<void> logout() async {
    await ApiService.clearTokens();
  }
}